react-components/src/components/Architecture/pointsOfInterest/PoIList.tsx

```diff
@@ -0,0 +1,59 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+import { type ReactElement, ReactNode, useState } from 'react';
+import { type PointsOfInterestDomainObject } from '../../../architecture/concrete/pointsOfInterest/PointsOfInterestDomainObject';
+import { type PointOfInterest } from '../../../architecture/concrete/pointsOfInterest/types';
+import { useOnUpdateDomainObject } from '../useOnUpdate';
+import { DataGrid, DatagridColumn } from '@cognite/cogs-lab';
+import { usePoIDomainObject } from './usePoIDomainObject';
+import { EMPTY_ARRAY } from '../../../utilities/constants';
+import { useTranslation } from '../../i18n/I18n';
+
+type RowType = {
+  id: string;
+  poi: PointOfInterest<any>;
+};
+
+export const PoIList = (): ReactNode => {
+  const { t } = useTranslation();
+
+  const [pois, setPois] = useState<Array<PointOfInterest<any>>>([]);
+  const [selectedPoi, setSelectedPoi] = useState<PointOfInterest<any> | undefined>(undefined);
+
+  const poiObject = usePoIDomainObject();
+
+  useOnUpdateDomainObject(poiObject, () => {
+    setPois([...(poiObject?.pointsOfInterest ?? EMPTY_ARRAY)]);
+    setSelectedPoi(poiObject?.selectedPointsOfInterest);
+  });
+
+  const columns: DatagridColumn<RowType>[] = [
+    {
+      field: 'name',
+      headerName: t('NAME', 'Name'),
+      flex: 3
+    }
+  ];
+
+  if (poiObject === undefined) {
+    return undefined;
+  }
+
+  return (
+    <DataGrid<RowType>
+      onRowClick={(row) => {
+        poiObject.setSelectedPointOfInterest(row.row.poi as PointOfInterest<any>);
```

Comment:
I was thinking if we can get rid of the casting and the any (unknown perhaps?) but this is not that big deal for now, IMO.

Reply:
Seems like it wasn't that easy :sweat: I wish the DataGrid component had proper typing on `onRowClick`, but it hasn't, for some reason.. I can put `unknown`, though :+1: